numpyro/distributions/continuous.py

```diff
@@ -1188,6 +1192,145 @@ def infer_shapes(
         return batch_shape, event_shape
 
 
+class CAR(Distribution):
+    r"""
+    The Conditional Autoregressive (CAR) distribution is a special case of the multivariate
+    normal in which the precision matrix is structured according to the adjacency matrix of
+    sites. The amount of autocorrelation between sites is controlled by :math:`\alpha`. The
+    distribution is a popular prior for areal spatial data.
+
+    :param float or ndarray loc: mean of the multivariate normal
+    :param float alpha: autoregression parameter. For most cases, the value should lie
+        between 0 (sites are independent, collapses to an iid multivariate normal) and
+        1 (perfect autocorrelation between sites), but the specification allows for negative
+        correlations.
+    :param float tau: positive precision for the multivariate normal
+    :param numpy.ndarray or scipy.sparse.csr_matrix W: symmetric adjacency matrix where 1
+        indicates adjacency between sites and 0 otherwise
+    :param bool is_sparse: whether to use a sparse form of W in calculations (must be True is
+        W is a :class:`scipy.sparse.spmatrix`)
+    """
+
+    arg_constraints = {
+        "loc": constraints.real_vector,
+        "alpha": constraints.open_interval(-1, 1),
+        "tau": constraints.positive,
+        "W": constraints.dependent_property(is_discrete=False, event_dim=2),
+    }
+    support = constraints.real_vector
+    reparametrized_params = [
+        "loc",
+        "alpha",
+        "tau",
+        "W",
+    ]
+
+    def __init__(self, loc, alpha, tau, W, *, is_sparse=False, validate_args=None):
+        if jnp.ndim(loc) == 0:
+            (loc,) = promote_shapes(loc, shape=(1,))
+
+        self.is_sparse = is_sparse
+
+        if self.is_sparse:
+            if not sparse.issparse(W):
+                assert isinstance(W, np.ndarray)
```

Comment:
maybe `assert sparse.issparse(W) or (isinstance(W, np.ndarray), W.ndim == 2)` is clearer